src/signTx.c

```diff
@@ -649,17 +649,51 @@ static void signTx_handleAuxDataAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_t
 // ============================== INPUTS ==============================
 
 enum {
-	HANDLE_INPUT_STEP_RESPOND = 200,
+	HANDLE_INPUT_STEP_SHOW_TX_HASH = 200,
+	HANDLE_INPUT_STEP_SHOW_INDEX,
+	HANDLE_INPUT_STEP_RESPOND,
 	HANDLE_INPUT_STEP_INVALID,
 };
 
+__noinline_due_to_stack__
 static void signTx_handleInput_ui_runStep()
 {
 	TRACE("UI step %d", ctx->ui_step);
 	ui_callback_fn_t* this_fn = signTx_handleInput_ui_runStep;
 
+	sign_tx_input_data_t* input = &txBodyCtx->stageData.input;
+
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
+	UI_STEP(HANDLE_INPUT_STEP_SHOW_TX_HASH) {
+		char header[30];
+		explicit_bzero(header, SIZEOF(header));
+		snprintf(header, SIZEOF(header), "Input %u: tx id", txBodyCtx->currentInput + 1);
+		ASSERT(strlen(header) < SIZEOF(header));
```

Comment:
We don't trust snprintf this much?

Reply:
We do, but this strict inequality ensures that the allocated buffer size is enough to include '\0'.

Reply:
snprintf does that already.

> Maximum number of bytes to be used in the buffer.
> The generated string has a length of at most n-1, leaving space for the additional terminating null character.
size_t is an unsigned integral type.

Reply:
Hm, good to know. It really does, I checked nanos-secure-sdk repo. But the code of snprintf also contains
```
                        // skip '-' still buggy ...
```
so I am wary of removing the check, it does not really hurt to keep it.

Reply:
Ah, but it does not, for buffers of size 1  --- so no, we don't fully trust snprintf  :D
```
    if(str == NULL ||str_size < 2) {
      return 0;
    }

```

Reply:
This snprintf implementation is so weird. And on the website they say to use C11 features and how advanced they are, while the SDK lacks these random things.

Reply:
There they go with C11 features:
https://github.com/LedgerHQ/app-cardano/pull/16
Using gnu11 produced clang analyzer warnings (tons of the same kind in nanos sdk repo mostly), so I asked them what can be done about that, and... this is the result :)